tidy3d/components/monitor.py

```diff
@@ -638,10 +645,23 @@ class ModeMonitor(AbstractModeMonitor):
         "and is hard-coded for other monitors depending on their specific function.",
     )
 
+    direction: Direction = pydantic.Field(
+        "+",
+        title="Propagation Direction",
+        description="Direction of propagation along the normal axis of the monitor used for the "
+        "computed mode field profiles.",
+    )
```

Comment:
I understand why you've done this but I think it may be confusing to have this argument but still return mode amps in both directions. Like, someone reading the api docs may think that they have to set to "-" if they're doing mode decomposition in the backward direction. It's not terrible but it's a bit annoying to have this inconsistency (`direction` only affecting the mode fields but not the mode amps). However, I do agree that the other options that I see are also not great:
- Do not provide `direction` and store fields in both directions, just like amps
- Store amps only for the specified `direction`

So I don't know, maybe at least we should call it `store_fields_direction`? Or how about make `store_fields` a spec, `store_fields = StoreFieldsSpec(direction="+")`? We could imagine extending this in the future to e.g. only request a subset of frequencies, like just the central freq, etc.

Reply:
Yes, I considered the same options when implementing this way but storing fields in both directions would be a major change and make `ModeData` and `ModeSolverData` interfaces incompatible.

I think your idea of using `store_fields` with a spec might be the better solution here. I'll go with that for now.

Reply:
After implementing the `StoreFieldsSpec` I think we should reconsider. This looks a lot like over-engineering to me. Maybe using `store_fields_direction` makes more sense. I reverted the changes to this simpler implementation but left both in the commit history just in case.

Frequency and multiple direction selection are rare use cases, and can be accomplished by setting up multiple monitors. That way we keep the usual use case straightforward, but the advanced uses still possible.

Reply:
Yeah, I agree with this evaluation.